training/COMPARISON.md

```diff
@@ -0,0 +1,462 @@
+3   Jake  FromStateFarm   32   35000
+```
+
+<a id="df_ak_create"></a>
+### Creating Arkouda DataFrame
+Arkouda DataFrames require the columns to be `pdarray` objects. The dictionary structure is below:
+```json
+{
+   "column_name_1": ak.array([0, 1, 2]),
+   "column_name_2": ak.array(["a", "b", "c"])
+}
+```
+
+```python
+import arkouda as ak
+ak.connect()
+ak_df = ak.DataFrame({
+    'F_Name': ak.array(['John', 'Jane', 'John', 'Jake']),
+    'L_Name': ak.array(['Doe', 'Doe', 'Smith', 'FromStateFarm']),
+    'Age': ak.array([37, 35, 50, 32]),
+    'Salary': ak.array([75000, 77000, 100000, 35000])
+})
+ak_df
+  F_Name         L_Name  Age  Salary
+0   John            Doe   37   75000
+1   Jane            Doe   35   77000
+2   John          Smith   50  100000
+3   Jake  FromStateFarm   32   35000 (4 rows x 4 columns)
+```
+
+<a id="df_groupby"></a>
+### GroupBy on DataFrames
+`GroupBy` on DataFrames is similar in concept to `GroupBy` on arrays. When calling `.count()` on the resulting `GroupBy` object there is some variance in the return from Arkouda and `Pandas`. 
+
+#### GroupBy on Pandas DataFrame
+One important note is that Pandas support grouping on both axes. Arkouda does not currently support this.
+```python
+import pandas as pd
+pd_df = pd.DataFrame({
+    'F_Name': ['John', 'Jane', 'John', 'Jake'],
+    'L_Name': ['Doe', 'Doe', 'Smith', 'FromStateFarm'],
+    'Age': [37, 35, 50, 32],
+    'Salary': [75000, 77000, 100000, 35000]
+})
+pd_g = pd_df.groupby("F_Name")
+pd_g.count()
+        L_Name  Age  Salary
+F_Name                     
+Jake         1    1       1
+Jane         1    1       1
+John         2    2       2
+```
+Notice that the return of count is a `DataFrame` object with each value representing the number of different values in a given column for the corresponding grouped column.
+
+#### GroupBy on Arkouda DataFrame
+Running `GroupBy` on and Arkouda `DataFrame` results in an Arkouda `GroupBy` object. This results in different output when running `.count()` on the result.
+```python
+import arkouda as ak
+ak.connect()
+ak_df = ak.DataFrame({
+    'F_Name': ak.array(['John', 'Jane', 'John', 'Jake']),
+    'L_Name': ak.array(['Doe', 'Doe', 'Smith', 'FromStateFarm']),
+    'Age': ak.array([37, 35, 50, 32]),
+    'Salary': ak.array([75000, 77000, 100000, 35000])
+})
+ak_g = ak_df.GroupBy("F_Name")
+ak_g.count()
+(array(['John', 'Jane', 'Jake']), array([2 1 1]))
+```
+Notice the return here is the unique keys found in the column requested to group on along with the number of time each of those keys occured in the column. 
+
+<a id="ex"></a>
+## Example Application
+Now that we have highlighted some of Arkouda's key components, let's walk through an example of how to use them. A good example of this is walking through the code for a set operation on a sequence of `pdarray` objects. We will go through one of these functions, `setdiff1d`. The goal here is to identify the values only found in the first object provided.
```

Comment:
perhaps we could give a high level explanation of `setdiff` in case the reader isn't familiar. I think just giving a quick example should be enough

Given two sets `s1, s2`, the setdiff (`s1-s2`) consists of the elements in `s1` that are not contained in `s2`. For example:
```
s1 = {1,2,3,4}
s2 = {3,7,8,2}
s1-s2 = {1,4}
```

Reply:
Personally, I believe that is easy enough to google. We aren't really here to explain the mathematical concepts in my opinion. But I'll add it.